compiler/rustc_builtin_macros/src/errors.rs

```diff
@@ -643,20 +654,29 @@ pub(crate) struct FormatUnknownTrait<'a> {
     pub(crate) suggs: Vec<FormatUnknownTraitSugg>,
 }
 
-#[derive(Subdiagnostic)]
-#[suggestion(
-    builtin_macros_suggestion,
-    code = "{fmt}",
-    style = "tool-only",
-    applicability = "maybe-incorrect"
-)]
 pub(crate) struct FormatUnknownTraitSugg {
-    #[primary_span]
     pub span: Span,
     pub fmt: &'static str,
     pub trait_name: &'static str,
 }
 
+impl rustc_errors::Subdiagnostic for FormatUnknownTraitSugg {
+    fn add_to_diag<G: rustc_errors::EmissionGuarantee>(self, diag: &mut rustc_errors::Diag<'_, G>) {
+        diag.arg("fmt", self.fmt);
+        diag.arg("trait_name", self.trait_name);
+        let msg = diag.eagerly_translate(crate::fluent_generated::builtin_macros_suggestion);
+        diag.remove_arg("fmt");
+        diag.remove_arg("trait_name");
```

Comment:
afaict the derive also eagerly translates, so we could just change the derive to also remove all those args again after the calls